master/docs/manual/worker-transition.rst

```diff
@@ -0,0 +1,1091 @@
+     - :py:meth:`buildbot.status.base.StatusReceiverBase.workerUnpaused`
+
+   * - :py:meth:`buildbot.interfaces.IStatus.getSlaveNames`
+       with all implementations
+     - :py:meth:`buildbot.interfaces.IStatus.getWorkerNames`
+
+   * - :py:meth:`buildbot.interfaces.IStatus.getSlave`
+       with all implementations
+     - :py:meth:`buildbot.interfaces.IStatus.getWorker`
+
+
+   * - :py:meth:`buildbot.interfaces.IBuildStatus.getSlavename`
+       with all implementations
+     - :py:meth:`buildbot.interfaces.IBuildStatus.getWorkername`
+
+   * - :py:meth:`buildbot.status.build.BuildStatus.setSlavename`
+     - :py:meth:`buildbot.status.build.BuildStatus.setWorkername`
+
+   * - :py:attr:`buildbot.status.build.BuildStatus.slavename`
+     - :py:attr:`buildbot.status.build.BuildStatus.workername`
+       (also it was moved from class static attribute to instance attribute)
+
+
+   * - :py:meth:`buildbot.interfaces.IBuilderStatus.getSlaves`
+       with all implementations
+     - :py:meth:`buildbot.interfaces.IBuilderStatus.getWorkers`
+
+   * - :py:attr:`buildbot.status.builder.BuilderStatus.slavenames`
+     - :py:attr:`buildbot.status.builder.BuilderStatus.workernames`
+
+   * - :py:meth:`buildbot.status.builder.BuilderStatus.setSlavenames`
+     - :py:meth:`buildbot.status.builder.BuilderStatus.setWorkernames`
+
+
+   * - :py:meth:`buildbot.process.botmaster.BotMaster.slaveLost`
+     - :py:meth:`buildbot.process.botmaster.BotMaster.workerLost`
+
+   * - :py:meth:`buildbot.process.botmaster.BotMaster.getBuildersForSlave`
+     - :py:meth:`buildbot.process.botmaster.BotMaster.getBuildersForWorker`
+
+   * - :py:meth:`buildbot.process.botmaster.BotMaster.maybeStartBuildsForSlave`
+     - :py:meth:`buildbot.process.botmaster.BotMaster.maybeStartBuildsForWorker`
+
+
+   * - :py:class:`buildbot.locks.RealSlaveLock`
+     - :py:class:`buildbot.locks.RealWorkerLock`
+
+   * - :py:attr:`buildbot.locks.RealSlaveLock.maxCountForSlave`
+     - :py:attr:`buildbot.locks.RealWorkerLock.maxCountForWorker`
+
+Other changes (done without providing fallback):
+
+* Functions argument ``buildslaveName`` renamed to ``workerName``.
+
+* Loop variables, local variables, helper functions:
+
+  .. list-table::
+     :header-rows: 1
+
+     * - Old name
+       - New name
+
+     * - ``s``
+       - ``w`` or ``worker``
+
+     * - ``sl``
+       - ``w`` or ``worker``
+
+     * - ``bs`` ("buildslave")
+       - ``w``
+
+     * - ``sb``
+       - ``wfb`` ("worker for builder")
+
+     * - ``bs1()``, ``bs2()``
+       - ``w1()``, ``w2()``
+
+     * - ``bslave``
+       - ``worker``
+
+     * - ``BS1_NAME``, ``BS1_ID``, ``BS1_INFO``
+       - ``W1_NAME``, ``W1_ID``, ``W1_INFO``
+
+* In :py:meth:`buildbot.config.BuilderConfig.getConfigDict` result
+  ``'slavenames'`` key changed to ``'workernames'``;
+  ``'slavebuilddir'`` key changed to ``'workerbuilddir'``;
+  ``'nextSlave'`` key changed to ``'nextWorker'``.
```

Comment:
Can you provide example of this class or `getConfigDict()` usage in configs?
I though that in config only plain Python dict is used, e.g.:

```
c = BuildmasterConfig = {}
```